### PR TITLE
Jerryliu sc 81984 update loginwithemailotp interface with new

### PR DESCRIPTION
**Pull request by Ethella**
### 📦 Pull Request

[Provide a general summary of the pull request here.]

### ✅ Fixed Issues

- [List any fixed issues here like: Fixes #XXXX]

### 🚨 Test instructions

[Describe any additional context required to test the PR/feature/bug fix.]

### ⚠️ Don't forget to add a [semver](https://semver.org/) label! 
Please only add **one** label:
- `patch`: Bug Fix?
- `minor`: New Feature?
- `major`: Breaking Change?
- `skip-release`: It's unnecessary to publish this change.

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @magic-ext/algorand@14.1.0-canary.596.5801506958.0
  npm install @magic-ext/aptos@2.1.0-canary.596.5801506958.0
  npm install @magic-ext/auth@2.1.0-canary.596.5801506958.0
  npm install @magic-ext/avalanche@14.1.0-canary.596.5801506958.0
  npm install @magic-ext/bitcoin@14.1.0-canary.596.5801506958.0
  npm install @magic-ext/conflux@12.1.0-canary.596.5801506958.0
  npm install @magic-ext/cosmos@14.1.0-canary.596.5801506958.0
  npm install @magic-ext/ed25519@10.1.0-canary.596.5801506958.0
  npm install @magic-ext/flow@14.1.0-canary.596.5801506958.0
  npm install @magic-ext/gdkms@2.1.0-canary.596.5801506958.0
  npm install @magic-ext/harmony@14.1.0-canary.596.5801506958.0
  npm install @magic-ext/icon@14.1.0-canary.596.5801506958.0
  npm install @magic-ext/near@14.1.0-canary.596.5801506958.0
  npm install @magic-ext/oauth@13.1.0-canary.596.5801506958.0
  npm install @magic-ext/oidc@2.1.0-canary.596.5801506958.0
  npm install @magic-ext/polkadot@14.1.0-canary.596.5801506958.0
  npm install @magic-ext/react-native-bare-oauth@14.1.0-canary.596.5801506958.0
  npm install @magic-ext/react-native-expo-oauth@14.1.0-canary.596.5801506958.0
  npm install @magic-ext/solana@15.1.0-canary.596.5801506958.0
  npm install @magic-ext/taquito@11.1.0-canary.596.5801506958.0
  npm install @magic-ext/terra@11.1.0-canary.596.5801506958.0
  npm install @magic-ext/tezos@14.1.0-canary.596.5801506958.0
  npm install @magic-ext/webauthn@13.1.0-canary.596.5801506958.0
  npm install @magic-ext/zilliqa@14.1.0-canary.596.5801506958.0
  npm install @magic-sdk/commons@15.1.0-canary.596.5801506958.0
  npm install @magic-sdk/pnp@13.1.0-canary.596.5801506958.0
  npm install @magic-sdk/provider@19.1.0-canary.596.5801506958.0
  npm install @magic-sdk/react-native-bare@20.1.0-canary.596.5801506958.0
  npm install @magic-sdk/react-native-expo@20.1.0-canary.596.5801506958.0
  npm install @magic-sdk/types@16.1.0-canary.596.5801506958.0
  npm install magic-sdk@19.1.0-canary.596.5801506958.0
  # or 
  yarn add @magic-ext/algorand@14.1.0-canary.596.5801506958.0
  yarn add @magic-ext/aptos@2.1.0-canary.596.5801506958.0
  yarn add @magic-ext/auth@2.1.0-canary.596.5801506958.0
  yarn add @magic-ext/avalanche@14.1.0-canary.596.5801506958.0
  yarn add @magic-ext/bitcoin@14.1.0-canary.596.5801506958.0
  yarn add @magic-ext/conflux@12.1.0-canary.596.5801506958.0
  yarn add @magic-ext/cosmos@14.1.0-canary.596.5801506958.0
  yarn add @magic-ext/ed25519@10.1.0-canary.596.5801506958.0
  yarn add @magic-ext/flow@14.1.0-canary.596.5801506958.0
  yarn add @magic-ext/gdkms@2.1.0-canary.596.5801506958.0
  yarn add @magic-ext/harmony@14.1.0-canary.596.5801506958.0
  yarn add @magic-ext/icon@14.1.0-canary.596.5801506958.0
  yarn add @magic-ext/near@14.1.0-canary.596.5801506958.0
  yarn add @magic-ext/oauth@13.1.0-canary.596.5801506958.0
  yarn add @magic-ext/oidc@2.1.0-canary.596.5801506958.0
  yarn add @magic-ext/polkadot@14.1.0-canary.596.5801506958.0
  yarn add @magic-ext/react-native-bare-oauth@14.1.0-canary.596.5801506958.0
  yarn add @magic-ext/react-native-expo-oauth@14.1.0-canary.596.5801506958.0
  yarn add @magic-ext/solana@15.1.0-canary.596.5801506958.0
  yarn add @magic-ext/taquito@11.1.0-canary.596.5801506958.0
  yarn add @magic-ext/terra@11.1.0-canary.596.5801506958.0
  yarn add @magic-ext/tezos@14.1.0-canary.596.5801506958.0
  yarn add @magic-ext/webauthn@13.1.0-canary.596.5801506958.0
  yarn add @magic-ext/zilliqa@14.1.0-canary.596.5801506958.0
  yarn add @magic-sdk/commons@15.1.0-canary.596.5801506958.0
  yarn add @magic-sdk/pnp@13.1.0-canary.596.5801506958.0
  yarn add @magic-sdk/provider@19.1.0-canary.596.5801506958.0
  yarn add @magic-sdk/react-native-bare@20.1.0-canary.596.5801506958.0
  yarn add @magic-sdk/react-native-expo@20.1.0-canary.596.5801506958.0
  yarn add @magic-sdk/types@16.1.0-canary.596.5801506958.0
  yarn add magic-sdk@19.1.0-canary.596.5801506958.0
  ```
</details>

### CODE REVIEW

**Review by Dizigen — APPROVED** [review]

File: packages/@magic-sdk/provider/src/modules/auth.ts [hunk 1]
```diff
@@ -81,29 +83,26 @@ export class AuthModule extends BaseModule {
...
-      method: 'auth.loginWithEmailOTP()',
-      removalVersions: ProductConsolidationMethodRemovalVersions,
-      useInstead: '@magic-ext/auth auth.loginWithEmailOTP()',
-    }).log();
-    const { email, showUI } = configuration;
+    const { email, showUI, deviceCheckUI } = configuration;
     const requestPayload = createJsonRpcRequestPayload(
       this.sdk.testMode ? MagicPayloadMethod.LoginWithEmailOTPTestMode : MagicPayloadMethod.LoginWithEmailOTP,
-      [{ email, showUI }],
+      [{ email, showUI, deviceCheckUI }],
```
Comment (Ariflo) [inline packages/@magic-sdk/provider/src/modules/auth.ts:89]: @Ethella - Given [our history](https://www.notion.so/magiclabs/Email-OTP-Modal-was-Hidden-on-Native-Mobile-SDK-s-Due-To-Magic-SDK-Update-301ccd9a74bb4de9abe63697ce4ba149?pvs=4) with this bit of code can you confirm this won't affect our native SDKs currently implementations of `LoginWithEmailOTP`?

**Review by Ariflo — CHANGES_REQUESTED** [review]
Can you confirm that this change will not affect our Native SDK's implementation of `loginWithEmailOTP`? I'm a little traumatized by the last time we messed with this code: https://www.notion.so/magiclabs/Email-OTP-Modal-was-Hidden-on-Native-Mobile-SDK-s-Due-To-Magic-SDK-Update-301ccd9a74bb4de9abe63697ce4ba149?pvs=4

**Comment by Ethella** [comment]
Can you confirm that this change will not affect our Native SDK's implementation of loginWithEmailOTP? I'm a little traumatized by the last time we messed with this code: https://www.notion.so/magiclabs/Email-OTP-Modal-was-Hidden-on-Native-Mobile-SDK-s-Due-To-Magic-SDK-Update-301ccd9a74bb4de9abe63697ce4ba149?pvs=4

Yes, thx for bringing that up. The deviceCheckUI is expected to be nullable and will default to true. So logically, it won't affect mobile SDK. I'll make sure it's been tested once we ship the change in auth relayer

**Review by Ariflo — APPROVED** [review]
WFM 👍

**Pull request merged**